src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -178,7 +178,11 @@ internal DateTimeOffset GetLastAccessTime(ReadOnlySpan<char> path, bool continue
         }
 
         internal void SetLastAccessTime(string path, DateTimeOffset time)
-            => SetAccessWriteTimes(path, time.ToUnixTimeSeconds(), null);
+        {
+            long secs = time.ToUnixTimeSeconds();
+            long milli = time.ToUnixTimeMilliseconds() - secs * 1000;
```

Comment:
This is confusing. Why isn't this `nanoseconds` and `* 1_000_000` here?

Reply:
do you mean ```(time.ToUnixTimeMilliseconds() - secs * 1000) * 1_000_000 ;```